### PR TITLE
make managerEntries be loaded as ESM, for improved tree-shaking

### PR DESCRIPTION
**Pull request by ndelangen**
Issue:

I noticed that the whole preview-api is bundled into every manager entry (`manager.mjs`).
I was able to assert that this is because the core-server resolves the entry fully. (to prevent pnp issues).
But by resolving it, it resolves the `cjs` entry.

## What I did

I changed the code in `core-server` so it removes the file extension.
Then I configured the manager-builder to try adding `.mjs` to the path.
This is not a perfect solution.
A better solution would be to somehow use the exportsMap.

I also configured esbuild to optimize the code.

before:
Screenshot 2022-12-02 at 20 20 00

after:
Screenshot 2022-12-02 at 19 38 54

### CODE REVIEW

**Review by valentinpalkovic — APPROVED** [review]

File: code/lib/core-common/src/presets.ts [hunk 1]
```diff
@@ -113,11 +119,19 @@ export const resolveAddonName = (
     const managerEntries = [];
 
     if (managerFile) {
-      managerEntries.push(managerFile);
+      // we remove the extension
+      // this is a bit of a hack to try en be able to find .mjs files
```
Comment (shilman) [inline code/lib/core-common/src/presets.ts:123]:
```suggestion
      // this is a bit of a hack to try to find .mjs files
```

File: code/lib/core-common/src/presets.ts [hunk 2]
```diff
@@ -113,11 +119,19 @@ export const resolveAddonName = (
...
+      // node can only ever resolve .js files; it does not look at the exports field in package.json
+      const { dir: fdir, name: fname } = parse(managerFile);
+      managerEntries.push(join(fdir, fname));
     }
     // register file is the old way of registering addons
     if (!managerFile && registerFile && !presetFile) {
-      managerEntries.push(registerFile);
+      // we remove the extension
+      // this is a bit of a hack to try en be able to find .mjs files
```
Comment (shilman) [inline code/lib/core-common/src/presets.ts:131]:
```suggestion
      // this is a bit of a hack to try to find .mjs files
```

File: code/lib/core-common/src/presets.ts [hunk 3]
```diff
@@ -66,14 +67,19 @@ export const resolveAddonName = (
...
-    if (name.match(/\/(manager|register(-panel)?)(\.(js|ts|tsx|jsx))?$/)) {
+    const { dir: fdir, name: fname } = parse(resolved);
+
+    if (name.match(/\/(manager|register(-panel)?)(\.(js|mjs|ts|tsx|jsx))?$/)) {
       return {
         type: 'virtual',
         name,
-        managerEntries: [resolved],
+        // we remove the extension
+        // this is a bit of a hack to try en be able to find .mjs files
```
Comment (shilman) [inline code/lib/core-common/src/presets.ts:77]:
```suggestion
        // this is a bit of a hack to try to find .mjs files
```

**Review by shilman — APPROVED** [review]
LGTM!

**Comment by ndelangen** [comment]
arg, github only showed me your review comments after I merged.

**Pull request merged**